tests/succeed/format-deref/simple.fathom

```diff
@@ -0,0 +1,5 @@
+{
+  start <- stream_pos,
+  len <- deref start u16be,
+  data <- array16 len u16be,
+}
```

Comment:
Implies that the length of the array is also its first element?

Reply:
Ahh yeah there should be a reserved thingo there